<launch>

    <!-- Posix SITL environment launch script -->
    <arg name="x" default="0"/>
    <arg name="y" default="0"/>
    <arg name="z" default="0"/>
    <arg name="R" default="0"/>
    <arg name="P" default="0"/>
    <arg name="Y" default="0"/>

    <arg name="x2" default="-1"/>
    <arg name="y2" default="-3"/>

  <arg name="x3" default="-3"/>
    <arg name="y3" default="-4"/>


    <arg name="est" default="lpe"/>
    <arg name="vehicle" default="f450"/>
    <arg name="vehicle2" default="f450-1"/>
    <arg name="vehicle3" default="f450-2"/>

    <arg name="world" default="$(find mavlink_sitl_gazebo)/worlds/empty.world"/>
    <arg name="sdf" default="$(find mavlink_sitl_gazebo)/models/$(arg vehicle)/$(arg vehicle).sdf"/>
    <arg name="sdf2" default="$(find mavlink_sitl_gazebo)/models/$(arg vehicle2)/$(arg vehicle2).sdf"/>
    <arg name="sdf3" default="$(find mavlink_sitl_gazebo)/models/$(arg vehicle3)/$(arg vehicle3).sdf"/>
    
<arg name="rcS" default="$(find px4)/posix-configs/SITL/init/$(arg est)/$(arg vehicle)"/>
    <arg name="rcS2" default="$(find px4)/posix-configs/SITL/init/$(arg est)/$(arg vehicle2)"/>
    <arg name="rcS3" default="$(find px4)/posix-configs/SITL/init/$(arg est)/$(arg vehicle3)"/>


    <arg name="headless" default="false"/>
    <arg name="gui" default="true"/>
    <arg name="ns" default="/"/>

    <arg name="debug" default="false"/>
    <arg name="verbose" default="false"/>
    <arg name="paused" default="false"/>

    <arg name="log_output" default="screen" />

    <arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
    <arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />

    <node name="sitl" pkg="px4" type="px4" output="screen"
        args="$(find px4) $(arg rcS)">
    </node>

    <node name="sitl2" pkg="px4" type="px4" output="screen"
        args="$(find px4) $(arg rcS2)">
    </node>
 <node name="sitl3" pkg="px4" type="px4" output="screen"
        args="$(find px4) $(arg rcS3)">
    </node>


    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="headless" value="$(arg headless)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="world_name" value="$(arg world)" />
        <arg name="debug" value="$(arg debug)" />
        <arg name="verbose" value="$(arg verbose)" />
        <arg name="paused" value="$(arg paused)" />
    </include>

    <node name="$(anon vehicle_spawn)" output="screen" pkg="gazebo_ros" type="spawn_model"
        args="-sdf -file $(arg sdf) -model $(arg vehicle) -x $(arg x) -y $(arg y) -z $(arg z) -R $(arg R) -P $(arg P) -Y $(arg Y)"/>-->

      <node name="$(anon vehicle_spawn_2)" output="screen" pkg="gazebo_ros" type="spawn_model"
        args="-sdf -file $(arg sdf2) -model $(arg vehicle2) -x $(arg x2) -y $(arg y2) -z $(arg z) -R $(arg R) -P $(arg P) -Y $(arg Y)"/>

 <node name="$(anon vehicle_spawn_3)" output="screen" pkg="gazebo_ros" type="spawn_model"
        args="-sdf -file $(arg sdf3) -model $(arg vehicle3) -x $(arg x3) -y $(arg y3) -z $(arg z) -R $(arg R) -P $(arg P) -Y $(arg Y)"/>


    <node pkg="mavros" type="mavros_node" name="mavros1" required="true" clear_params="true" output="$(arg log_output)">
		<param name="fcu_url" value="udp://:14540@localhost:14557"/>
		<param name="gcs_url" value="" />
		<param name="target_system_id" value="1" />
		<param name="target_component_id" value="1" />
		<param name="system_id" value="34" />
		<rosparam command="load" file="$(arg pluginlists_yaml)" />
		<rosparam command="load" file="$(arg config_yaml)" />
	</node>

    <node pkg="mavros" type="mavros_node" name="mavros2" required="true" clear_params="true" output="$(arg log_output)">
		<param name="fcu_url" value="udp://:14640@localhost:14657"/>
		<param name="gcs_url" value="" />
		<param name="target_system_id" value="2" />
		<param name="target_component_id" value="2" />
		<param name="system_id" value="48" />
		<!-- load blacklist, config -->
		<rosparam command="load" file="$(arg pluginlists_yaml)" />
		<rosparam command="load" file="$(arg config_yaml)" />
	</node>
 <node pkg="mavros" type="mavros_node" name="mavros3" required="true" clear_params="true" output="$(arg log_output)">
                <param name="fcu_url" value="udp://:14740@localhost:14757"/>
                <param name="gcs_url" value="" />
                <param name="target_system_id" value="3" />
                <param name="target_component_id" value="3" />
                <param name="system_id" value="56" />
                <!-- load blacklist, config -->
                <rosparam command="load" file="$(arg pluginlists_yaml)" />
                <rosparam command="load" file="$(arg config_yaml)" />
        </node>


</launch>

<!-- vim: set et ft=xml fenc=utf-8 ff=unix sts=0 sw=4 ts=4 : -->
